pkg/cluster/internal/providers/podman/provision.go

```diff
@@ -162,7 +162,19 @@ func commonArgs(cfg *config.Cluster) ([]string, error) {
 }
 
 func runArgsForNode(node *config.Node, clusterIPFamily config.ClusterIPFamily, name string, args []string) ([]string, error) {
-	volume, err := createAnonymousVolume(name)
+	// Pre-create anonymous volumes to enable specifying mount options
+	// during container run time
+	containerdVolume, err := createAnonymousVolume(name)
```

Comment:
for future cleanup: instead of N calls to this I think we should have a wrapper generate N volumes to distribute amongst mounts / basically abstract """"anonymous volumes""" to more closely map to what we're doing in docker.

I think it wouldn't be hard to make a little API around this that does the allocation in one call (and one error check) to make it less painful to add more of these.

Reply:
After this PR gets merged, I can help to create another one that refactor this code. 

My idea is to create an array/map in `constants.go` that keeps track of all the volumes with their mount point, then destroy them when destroying the cluster by looking up the very field.

Let me know if this sounds like a good idea.